python/semver-0.x.x.md

```diff
@@ -0,0 +1,134 @@
+# Semantic version rules and guidelines for 0.x.x releases
+
+We follow [semantic versioning](https://semver.org/), but we also have many
+projects at 0.x.x, which means no rules at all in semver. So far the convention
+was to keep patch release as bug fix releases (so 0.X.y), and minor releases
+could include new features and/or breaking changes.
+
+As we have bigger projects with many dependencies, it is becoming more
+challenging to update dependencies when we don't have any guarantees for 0.x.x
+versions. As we don't want to prematurely declare a library 1.0.0 (because that
+carries some expectations about completeness, not only about breaking changes),
+we need to agree on some guarantees for 0.x.x releases.
+
+The goal is to allow developers to move their libraries that are under heavy
+development fast, and yet make the life of users of those libraries easier,
+without ending in [dependency
+hell](https://en.wikipedia.org/wiki/Dependency_hell).
+
+In this document we use the words like MUST, MAY, etc. with the meaning defined
+by [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Patch versions guarantees
+
+Provide the same guarantees as minor versions after 1.0.0, paraphrasing semver
+on [minor versions](https://semver.org/#spec-item-7), patch version Z (0.y.Z):
+
+* MUST be incremented if new, backward compatible functionality is introduced
+  to the public API.
+* MUST be incremented if any public API functionality is marked as deprecated.
+* MAY be incremented if substantial new functionality or improvements are
+  introduced within the private code.
+* MAY include [patch level changes](https://semver.org/#spec-item-6):
+
+  - MUST be incremented if only backward compatible bug fixes are introduced.
+  - A bug fix is defined as an internal change that fixes incorrect behavior.
+
+Additionally:
+
+* Deprecated symbols MAY NOT be maintained, it is recommended to switch to
+  non-deprecated symbols as soon as possible.
+
+## Minor versions guarantees
+
+Provide the same guarantees as major versions after 1.0.0, paraphrasing semver
+on [major versions](https://semver.org/#spec-item-8), minor version X (0.Y.z):
```

Comment:
Seems like there is a typo. Should be X (0.X.z):

Reply:
Fixed.